tests/test-cases/Group1-Docker-Commands/1-38-Docker-Exec.md

```diff
@@ -46,12 +50,15 @@ None
 * step 7 The container should have an exit code of 0.
 
 # Exec Power Off test for long running Process
+## Purpose:
+This test is designed to test running exec's against a container running a process which would not naturally exit. Then after running many execs concurrently we explicitly stop the container. We should then handle the the execs in their varying states properly.
+
 ## Test Steps
 1. Pull an image that contains `/bin/top`. Busybox suffices here.
-2. Create a container running `/bin/top` to simulate a long running process.
+2. Create a container running `/bin/top` to simulate a long running process that should not exit on it's own.
 3. Run the container and detach from it.
 4. Start 20 simple exec operations in parallel against the detached container.
-5. Stop the container while the execs are still running in order to trigger the exec power off errors.
+5. Explicitly stop the container while the execs are still running in order to trigger the exec power off errors.
```

Comment:
I think there is a conceptual race between (4) and (5). You are attempting to ensure that some of the execs have been dispatched before stop takes effect, and that some are issued _during the stop process_.
It's ok to note that this test is stochastic and not expected to behave in the same manner every time.